packages/frontend/src/components/ProjectConnection/WarehouseForms/SnowflakeForm.tsx

```diff
@@ -108,43 +110,53 @@ const SnowflakeForm: FC<{
         throw new Error('Snowflake form is not used for this warehouse type');
     }
 
+    const savedAuthType =
+        savedProject?.warehouseConnection?.type === WarehouseTypes.SNOWFLAKE
+            ? savedProject?.warehouseConnection?.authenticationType
+            : undefined;
     const hasPrivateKey =
         savedProject !== undefined
-            ? savedProject?.warehouseConnection?.type ===
-                  WarehouseTypes.SNOWFLAKE &&
-              savedProject?.warehouseConnection?.authenticationType ===
-                  'private_key'
+            ? savedAuthType === SnowflakeAuthenticationType.PRIVATE_KEY
             : true;
+    const defaultAuthType = savedAuthType
+        ? savedAuthType
+        : isSsoEnabled
+        ? SnowflakeAuthenticationType.SSO
+        : hasPrivateKey
+        ? SnowflakeAuthenticationType.PRIVATE_KEY
+        : SnowflakeAuthenticationType.PASSWORD;
 
-    const authenticationType: string =
-        form.values.warehouse.authenticationType ??
-        (hasPrivateKey ? 'private_key' : 'password');
+    if (!form.isDirty()) {
```

Comment:
We need react to re-render if the default value is different than what we started with. This can be because results coming back from SSO auth.